onnxruntime/core/providers/cuda/cuda_provider_factory.cc

```diff
@@ -62,3 +62,21 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider_CUDA,
 
   return nullptr;
 }
+
+ORT_API_STATUS_IMPL(OrtApis::SetCurrentGpuDeviceId, _In_ int device_id) {
+  auto cuda_err = cudaSetDevice(device_id);
```

Comment:
>device_id [](start = 32, length = 9)

get the number of compute-capable devices by cudaGetDeviceCount, validate the given device_id and prompt error if not valid